frontend/src/commons/ShareDialog/ContactsShareList.js

```diff
@@ -13,20 +14,57 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const ContactsShareList = ({ addInvitation, removeInvitation, announces, announcers, isOrganizer }) => {
+const ContactsShareList = ({ addInvitation, removeInvitation, announces, announcers, isOrganizer, newInvitations }) => {
   const classes = useStyles();
   const translate = useTranslate();
   const { ids, data, loading, ...rest } = useListContext();
+
+  const [records, setRecords] = React.useState([]);
+
+  useEffect(() => {
+    setRecords(ids.reduce((acc, id) => {
+      const viewSwitchReadonly = announces.includes(data[id].describes)
+      const shareSwitchReadonly = announcers.includes(data[id].describes)
+      return [...acc, {
+        ...data[id],
+        viewSwitchReadonly,
+        shareSwitchReadonly,
+        canViewEvent: announces.includes(data[id].describes),
+        canShareEvent: announcers.includes(data[id].describes),
+      }]
+    }, []))
+  }, [
+    ids, data, announcers, announces
+  ])
+
+  const updateRecord = React.useCallback((record) => {
+    const newRecords = records.map((item) => {
+      if (item.id === record.id) {
+        return { ...record }
+      }
+      return item
+    })
+    setRecords(newRecords)
+  }, [records, setRecords])
+
   return (
     <List dense className={classes.list}>
-      {ids.map((id, i) => (
+      {isOrganizer && (
```

Comment:
It would be important that the "All contacts" share button is also available for non-organizers (in this case, we would see only one toggle)